File: src/main/java/nu/peg/slack/pt/service/ConnectionService.java
package nu.peg.slack.pt.service;

import nu.peg.slack.pt.api.transport.TransportApi;

import org.jvnet.hk2.annotations.Service;

import javax.inject.Inject;

@Service
public class ConnectionService {

    private TransportApi transportApi;

    @Inject
    public ConnectionService(TransportApi transportApi) {
        this.transportApi = transportApi;
    }
}

File: src/main/kotlin/nu/peg/slack/pt/service/ConnectionService.kt
package nu.peg.slack.pt.service

import nu.peg.slack.pt.api.slack.model.CommandPostData
import nu.peg.slack.pt.api.slack.model.SlackMessage
import nu.peg.slack.pt.api.transport.model.Connection
import nu.peg.slack.pt.model.ConnectionRequest

interface ConnectionService {

    fun handleCommand(commandData: CommandPostData)

    fun makeConnectionOverview(request: ConnectionRequest): SlackMessage

    fun makeConnectionDetail(connection: Connection): SlackMessage
}
